### PR TITLE
[APPINT-201] Improve generated file

### PR DESCRIPTION
**Pull request by MarcoCode**

### CODE REVIEW

File: lib/generators/connector/templates/home_controller.rb [hunk 1]
```diff
@@ -11,20 +11,20 @@ def update
       old_sync_state = organization.sync_enabled
 
       organization.synchronized_entities.keys.each do |entity|
-        organization.synchronized_entities[entity] = !!params["#{entity}"]
+        organization.synchronized_entities[entity] = !params[entity.to_s].nil?
```
Comment (Berardpi) [inline lib/generators/connector/templates/home_controller.rb]: You're sure about this one?

File: lib/generators/connector/templates/stylesheets/layout.sass [hunk 2]
```diff
@@ -30,15 +30,27 @@ html, body
...
   &.navbar-inverse
     font-weight: 300
     letter-spacing: 1px
 
     ul.nav.navbar-nav
-      text-transform: uppercase
+      text-transform: $navbar-text-transform
+      font-weight:    $navbar-font-weight
 
       a
```
Comment (Berardpi) [inline lib/generators/connector/templates/stylesheets/layout.sass]:
```
a
  font-size: $normal
  
  &:hover, &:focuse
    color: ..
    text-decoration: ..
```

File: lib/generators/connector/templates/stylesheets/variables.sass [hunk 3]
```diff
@@ -36,6 +38,17 @@ $small:                          11px
 $normal:                         13px
```
Comment (Berardpi) [inline lib/generators/connector/templates/stylesheets/variables.sass]: I know it's not yours but the var name is not ideal (does not make sense without the Text Size comment)

File: spec/factories.rb [hunk 4]
```diff
@@ -21,9 +20,9 @@
   factory :idmap, class: Maestrano::Connector::Rails::IdMap do
     sequence(:connec_id) { |n| "#{n}6798-ada6-te43#{n}" }
     connec_entity 'person'
-    sequence(:external_id) { |n| "#{n}4567ada66#{n}" } 
+    sequence(:external_id) { |n| "#{n}4567ada66#{n}" }
     external_entity 'contact'
-    last_push_to_external 2.day.ago
+    last_push_to_external 2.days
```
Comment (Berardpi) [inline spec/factories.rb]: ago?

File: template/factories.rb [hunk 5]
```diff
@@ -12,7 +11,7 @@
     connec_entity 'person'
     external_id '4567ada66'
     external_entity 'contact'
-    last_push_to_external 2.day.ago
+    last_push_to_external 2.days
```
Comment (Berardpi) [inline template/factories.rb]: ago ?

File: template/maestrano.rb [hunk 6]
```diff
@@ -140,7 +139,7 @@
     #
     # == Subscriptions
     # This is the list of entities (organizations,people,invoices etc.) for which you want to be
-    # notified upon creation/update in Connec!™
+    # notified upon creation/update in Connec! Copyright.
```
Comment (Berardpi) [inline template/maestrano.rb]: TM is rather trademark than copyright, but I think you can just put nothing here, we usually don't but the TM in the code

File: template/maestrano_connector_template.rb [hunk 7]
```diff
@@ -43,14 +43,14 @@ def source_paths
...
+  gem 'activerecord-jdbcpostgresql-adapter', platforms: :jruby
+  gem 'pg', platforms: :ruby
   gem 'rails_12factor'
 end
 
 gem_group :test, :develpment do
-  gem 'activerecord-jdbcsqlite3-adapter', :platforms => :jruby
-  gem 'sqlite3', :platforms => :ruby
+  gem 'activerecord-jdbcsqlite3-adapter', platforms: :jruby
+  gem 'sqlite3', platforms: :ruby
```
Comment (Berardpi) [inline template/maestrano_connector_template.rb]: can you add rubocop here?

File: .rubocop.yml [hunk 8]
```diff
@@ -12,8 +12,8 @@ AllCops:
     - 'db/migrate/*'
     - 'bin/*'
     - 'spec/**/*'
-    - 'template/**/*'
-    - 'lib/generators/connector/**/*'
+    - 'lib/generators/connector/templates/entity.rb'
```
Comment (Berardpi) [inline .rubocop.yml]: Why not those two?

File: lib/generators/connector/templates/home_controller.rb [hunk 9]
```diff
@@ -7,46 +8,35 @@ def index
...
-        end
-      end
-      
-      organization.save
-
-      if !old_sync_state && organization.sync_enabled
-        Maestrano::Connector::Rails::SynchronizationJob.perform_later(organization, {})
-        flash[:info] = 'Congrats, you\'re all set up! Your data are now being synced'
-      end
+    return unless organization && is_admin?(current_user, organization)
```
Comment (Berardpi) [inline lib/generators/connector/templates/home_controller.rb]: I'm being paranoid, but can double check that the spec are testing failing cases (no id, not admin) and are green (for all the methods of this controller)

File: template/rubocop [hunk 10]
```diff
@@ -0,0 +1,49 @@
+Rails:
```
Comment (Berardpi) [inline template/rubocop]: Can you rename this file to a .yml?

**Comment by Berardpi** [comment]
LGTM 👍 Will merge that next week
Sorry for the tmp/cache/.keep, I have not found a clean way to manage that yet.

File: lib/generators/connector/templates/home_controller.rb (hunk 9 above)
Comment (MarcoCode) [inline lib/generators/connector/templates/home_controller.rb]: You are not paranoid, I just fixed that

File: lib/generators/connector/templates/home_controller.rb [hunk 11]
```diff
@@ -1,52 +1,47 @@
+# frozen_string_literal: true
 class HomeController < ApplicationController
+  before_action :organization_to_update, only: [:update]
```
Comment (Berardpi) [inline lib/generators/connector/templates/home_controller.rb]: Isn't that a bit overkill to refactor one line used only once?

File: lib/generators/connector/templates/home_controller.rb [hunk 12]
```diff
@@ -1,52 +1,47 @@
...
+    return redirect_to(:back) unless @updating_organization && is_admin?(current_user, @updating_organization)
+    old_sync_state = @updating_organization.sync_enabled
+    @updating_organization.synchronized_entities.keys.each do |entity|
+      @updating_organization.synchronized_entities[entity] = params[entity.to_s].present?
     end
+    @updating_organization.sync_enabled = @updating_organization.synchronized_entities.values.any?
+    @updating_organization.check_historical_data(params['historical-data'].present?)
 
-    redirect_to(:back)
+    unable_to_start = start_synchronization(old_sync_state, @updating_organization)
```
Comment (Berardpi) [inline lib/generators/connector/templates/home_controller.rb]:
```ruby
start_synchronization(old_sync_state, @updating_organization) unless !old_sync_state && organization.sync_enabled
redirect_to(:back)
```

**Pull request merged**